module/zfs/dmu.c

```diff
@@ -860,6 +840,26 @@ dmu_free_long_range_impl(objset_t *os, dnode_t *dn, uint64_t offset,
 			return (err);
 		}
 
+		txg = dmu_tx_get_txg(tx);
+
+		mutex_enter(&dp->dp_lock);
+		long_free_dirty +=
+		    dp->dp_long_free_dirty_pertxg[txg & TXG_MASK];
+		mutex_exit(&dp->dp_lock);
+
+		/*
+		 * To avoid filling up a TXG with just frees, wait for
+		 * the next TXG to open before freeing more chunks if
+		 * we have reached the threshold of frees.
+		 */
+		if (dirty_frees_threshold != 0 &&
+		    long_free_dirty >= dirty_frees_threshold) {
+			DMU_TX_STAT_BUMP(dmu_tx_dirty_frees_delay);
+			dmu_tx_commit(tx);
+			txg_wait_open(dp, 0, B_TRUE);
+			continue;
+		}
```

Comment:
No objection, but a question.  Was this hunk moved down and the `dmu_tx_commit(tx)` added solely to improve the readability?

Reply:
No. it was moved down so this bit:
```
		mutex_enter(&dp->dp_lock);
		long_free_dirty +=
		    dp->dp_long_free_dirty_pertxg[txg & TXG_MASK];
		mutex_exit(&dp->dp_lock);
```
had access to the txg.